carbon-design-system/carbon
# fix(combobox): update to match ARIA 1.2 criteria

Closes #6938 

Updates the combobox implementation to match the [authoring practices reference](https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-list.html) and includes updates from #6938 as suggested by @carmacleod 

#### Changelog

**New**

- Add new versions of components for ListBox that more closely align with where we want these components to go, this helps us add the required behavior into combobox without breaking existing components

**Changed**

- Update ComboBox so that the wrapper node no longer has `getRootProps` applied to it, these attributes now are on the `input`
- Convert both the trigger and clear icons to buttons and remove them from tab order (the equivalent functionality already exists through keyboard)

**Removed**

#### Testing / Reviewing

- Verify only combobox is impacted by these changes (splash zone includes dropdown, multiselect, filterable multiselect)
- Verify AC has no violations with both closed and open menus
- Verify Combobox can be operated by VO, JAWS, and NVDA
- Verify the following changes have occurred to the markup:
  - [ ] Outermost node no longer has combobox role and is now a wrapper div
  - [ ] The list box field is no longer interactive and has no additional attributes like ARIA labels
  - [ ] The list box input, menu icon, and selection are now siblings instead of descendants of a field which is a button
  - [ ] Focus shifts to the input when the menu is open or closed
  - [ ] Focus shifts to the input when the input is cleared
  - [ ] Both the menu trigger and clear button are not in tab order
